src/Stratis.Bitcoin/P2P/Peer/NetworkPeerFactory.cs

```diff
@@ -22,8 +22,9 @@ public interface INetworkPeerFactory
         /// </summary>
         /// <param name="client">Already connected network client.</param>
         /// <param name="parameters">Parameters of the established connection, or <c>null</c> to use default parameters.</param>
+        /// <param name="connectedPeersHelper">Maintains a list of connected peers and ensures their proper disposal. Or <c>null</c> if case disposal should be handled from user code.</param>
         /// <returns>New network peer that is connected via the established connection.</returns>
-        INetworkPeer CreateNetworkPeer(TcpClient client, NetworkPeerConnectionParameters parameters = null);
+        INetworkPeer CreateNetworkPeer(TcpClient client, NetworkPeerConnectionParameters parameters = null, ConnectedPeersHelper connectedPeersHelper = null);
```

Comment:
Why to allow null helpers?

Reply:
In case disposal should be handled in the user code. 
Right now only peer discovery uses that.

Reply:
Ok I am struggling to grasp why the DisposeHelper needs to be an external component to the peer.

Reply:
As explained above, if the component does not want to track and dispose its peers, it can use the helper. This is optional however because at least one component in our code manage disposal itself and does not need to track peers at all.

It would probably better for me to make a Skype call to explain this design.